src/Data/PQueue/Internals.hs

```diff
@@ -603,6 +608,63 @@ instance Functor rk => Functor (BinomForest rk) where
   fmap f (Skip ts) = Skip (fmap f ts)
   fmap f (Cons t ts) = Cons (fmap f t) (fmap f ts)
 
+instance Foldr Zero where
+  foldr_ _ z ~Zero = z
```

Comment:
Btw, why are you using a lazy pattern match here?

Reply:
There's no reason to actually match on the `Zero` constructor, since it carries no information. The pattern is only there to make it obvious what is there and why we don't care about it.